pkgs/top-level/all-packages.nix

```diff
@@ -13873,7 +13873,13 @@ with pkgs;
     stdenv = gcc7Stdenv;
   }));
 
-  llvmPackages_latest = llvmPackages_13;
+  llvmPackages_latest = recurseIntoAttrs (callPackage ../development/compilers/llvm/git ({
```

Comment:
I think llvmPackages_latest should be the latest *stable* version, which is tested more than the latest git version.
For example mesa uses llvmPackages_latest as the shader compiler on amd gpus (so this could even break mesa when the llvm C api gets backwards-incompatible changes).

Adding a new llvmPackages_git for the experimental version sounds better to me.

Reply:
Agreed. I've moved `llvmPackages_git` from `top-level/aliases.nix` to `top-level/all-packages.nix`. I'm unsure if that is necessary, the comment above `llvmPackages_git` seems to indicate the package set is only for integration testing:
```
  # LLVM packages for (integration) testing that should not be used inside Nixpkgs:
  llvmPackages_git = recurseIntoAttrs (callPackage ../development/compilers/llvm/git {
    inherit (stdenvAdapters) overrideCC;
    buildLlvmTools = buildPackages.llvmPackages_git.tools;
    targetLlvmLibraries = targetPackages.llvmPackages_git.libraries;
  });
```
I can revert if necessary, leaving only the patch changes and top-level unchanged.

Reply:
Just some notes:

Yes, so far `llvmPackages_git` was intentionally in `aliases.nix` so that the CI tests fail if something in Nixpkgs starts to depend on it (which would likely start to happen but there should be few to none exceptions where `llvmPackages_git` is required). It also prevents Hydra from building `llvmPackages_git` (but I guess building it on Hydra should be fine).

Moving this to `all-packages.nix` should probably be fine as long as someone checks the reverse dependencies from time to time.

PS: But in general I'd recommend to put such (basically unrelated) changes into dedicated PRs as each additional change will IMO usually slow reviews down (at least if it requires additional discussion).

Reply:
Backed out of this for now.